src/templates/assets/javascripts/components/source/facts/gitlab/index.ts

```diff
@@ -49,13 +61,30 @@ export function fetchSourceFactsFromGitLab(
   base: string, project: string
 ): Observable<SourceFacts> {
   const url = `https://${base}/api/v4/projects/${encodeURIComponent(project)}`
-  return requestJSON<ProjectSchema>(url)
+  return zip(
+
+    /* Fetch version */
+    requestJSON<Release>(`${url}/releases/permalink/latest`)
```

Comment:
Does this also work with privately hosted GitLab? Or does it only work on GitLab's Cloud offering?

Reply:
With the current implementation, I would say no because those repositories are typically _private_, not public. However, the [API endpoint](https://docs.gitlab.com/ee/api/releases/) used is the same:

```
Tier: Free, Premium, Ultimate
Offering: GitLab.com, Self-managed, GitLab Dedicated
```

Reply:
Okay, we need to test if the additions break the stars/forks code, when run in a private installation. It's fine if the release isn't shown, but at least the stars should work. How can we ensure this?

Reply:
To complement my previous answer, the regular expression to detect whether a URL is from GitLab is not enough to cover all possible self-hosted GitLab instances. I also know [Framagit](https://framagit.org/explore) (by [Framasoft](https://framasoft.org/en/)) and the regular expression doesn't match:

https://github.com/squidfunk/mkdocs-material/blob/a5438a6dc23396023bbe9d73592fccd562ee4174/src/templates/assets/javascripts/components/source/facts/_/index.ts#L79-L80

However, the GitLab API endpoints work:

```bash
curl -L https:/framagit.org/api/v4/projects/framasoft%2Fmobilizon/releases/permalink/latest
```

I'm not sure how to resolve this (List of self-hosted GitLab instances to try to match a URL? API request?), so I'd say we can discuss it in another issue if you agree.

Reply:
Thanks for taking the time to clarify! I think my question was phrased a bit badly talking about privately hosted GitLab, where I just meant self-hosted. Of course this can't work with private repositories, it's the same for GitHub, so all good!

> To complement my previous answer, the regular expression to detect whether a URL is from GitLab is not enough to cover all possible self-hosted GitLab instances. 

Yes, we're aware of that, but we're currently not considering making configuration even more complicated. If works for the majority of cases, and otherwise can be forked and adjusted. This has been asked before. Since MkDocs is undergoing a [rewrite](https://github.com/mkdocs/sketch), we're currently not considering changing anything on the source functionality, since it is completely unclear how this functionality will be supported in the future. We'll do our absolute best to keep this feature, of course.

Reply:
Forgot to say: this is good to merge then!

Reply:
Thanks a lot, @squidfunk!